bin/generate_bulk_metadata.R

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env Rscript
+
+# This script generates the metadata.tsv file for bulk RNA-seq libraries.
+# It outputs one metadata file per project.
+
+# import libraries
+library(optparse)
+suppressPackageStartupMessages(library(SingleCellExperiment))
+
+# set up arguments
+option_list <- list(
+  make_option(
+    opt_str = c("-p", "--project_id"),
+    type = "character",
+    help = "scpca project ID",
+  ),
+  make_option(
+    opt_str = c("-s", "--salmon_dirs"),
+    type = "character",
+    help = "Path to text file containing salmon output directories, one per line."
+  ),
+  make_option(
+    opt_str = c("--library_metadata_file"),
+    type = "character",
+    help = "path to metadata file containing scpca_library_id, scpca_sample_id and associated metadata"
+  ),
+  make_option(
+    opt_str = "--metadata_output",
+    default = "metadata.tsv",
+    type = "character",
+    help = "path to metadata tsv output file"
+  ),
+  make_option(
+    opt_str = "--genome_assembly",
+    type = "character",
+    default = NA,
+    help = "genome assembly used for mapping"
+  ),
+  make_option(
+    opt_str = "--workflow_url",
+    type = "character",
+    default = NA,
+    help = "workflow github url"
+  ),
+  make_option(
+    opt_str = "--workflow_version",
+    type = "character",
+    default = NA,
+    help = "workflow version identifier"
+  ),
+  make_option(
+    opt_str = "--workflow_commit",
+    type = "character",
+    default = NA,
+    help = "workflow commit hash"
+  )
+)
+
+opt <- parse_args(OptionParser(option_list = option_list))
+
+# check for project id
+if(is.null(opt$project_id)){
+  stop("A `project_id` is required.")
+}
+
+
+# replace workflow url and commit if not provided
+if (is.null(opt$workflow_url)){
+  opt$workflow_url <- NA
+}
+if (is.null(opt$workflow_version)){
+  opt$workflow_version <- NA
+}
+if (is.null(opt$workflow_commit)){
+  opt$workflow_commit <- NA
+}
+
+# read in library metadata file 
+library_metadata <- readr::read_tsv(opt$library_metadata_file)
+
+# list of paths to salmon log files 
+library_ids <- readLines(opt$salmon_dirs)
+
+# subset library metadata to only contain libraries that are being processed 
+# only keep metadata columns of interest 
+bulk_metadata_df <- library_metadata |>
+  dplyr::filter(scpca_library_id %in% library_ids &
+                scpca_project_id %in% opt$project_id) |>
+  dplyr::select(scpca_sample_id, scpca_library_id, scpca_project_id,
+                technology, seq_unit) |> 
+  # rename column names to match format of metadata files from other modalities
+  dplyr::rename(
+    sample_id = scpca_sample_id,
+    library_id = scpca_library_id,
+    project_id = scpca_project_id
+  ) |>
+  # add columns with processing information and date processed (same for all libraries )
+  dplyr::mutate(date_processed = lubridate::format_ISO8601(lubridate::now(tzone = "UTC"), usetz = TRUE),
+                genome_assembly = opt$genome_assembly, 
```

Comment:
I don't love having the date set here, as this may not reflect when the actual processing happened. (I probably have the same complaint for the SC data, but that is set when the RDS is made, as I recall, so that may be closer to the actual processing time). Here I am anticipating that we might add skipping of the salmon mapping to bulk as well, so individual samples might have different processing times. I will put a suggestion in the `processing_info` section for how we might get what we want out.
```suggestion
  dplyr::mutate(genome_assembly = opt$genome_assembly, 
```